resources/build/conanfile.py

```diff
@@ -0,0 +1,44 @@
+from conans import ConanFile, CMake
+
+
+class OpenAssetIOConan(ConanFile):
+    generators = (
+        # Generate a CMake toolchain preamble file `conan_paths.cmake`,
+        # which augments package search paths with conan package
+        # directories.
+        "cmake_paths",
+        # Generate `Find<PackageName>.cmake` finders, required for
+        # various public ConanCenter packages (e.g. pybind11) since they
+        # disallow bundling of `<PackageName>Config.cmake`-like files in
+        # the package.
+        "cmake_find_package",
+    )
+    settings = "os"
+    options = {"require_test_deps": [True, False]}
+    # TODO(DF): For minimal friction, default enable install of unit
+    #  test libraries for now - we should switch this off by default
+    #  when we get to distributing a conan package.
+    default_options = {"require_test_deps": True}
+
+    def build_requirements(self):
+        # CY2022
+        self.tool_requires("cpython/3.9.7")
```

Comment:
Thanks for the detailed explanation! 

We have a parallel dev stream going on looking into building wheels on CI, where we have indeed used `cibuildwheel` and disabled the conan `cpython` requirement in that case, so that the external Python environment is used instead. We'll hopefully have a PR up for that within the next week or so.

Really interesting note on into the glibc version - I hadn't considered that.

Reply:
Added passing `Python_EXECUTABLE` CMake variable to `setup.py` in 8aaeef81.  Not technically necessary for this PR, but good practice.